src/mds/Server.cc

```diff
@@ -3057,6 +3057,20 @@ void Server::handle_client_getattr(MDRequestRef& mdr, bool is_lookup)
     }
   }
 
+  // non-auth MDS needs to sync latest rstat from auth MDS 
+  // if getattr request wants dir rstat, so try to acquire rdlock
+  // on nestlock to trigger the whole sync process
+  if (!ref->is_auth() && (mask & CEPH_STAT_RSTAT)) {
```

Comment:
I think about this again. Maybe it's better to just forward the request to auth mds.  this avoids rdlock nestlock.

Reply:
Yes, I already tried this way in the first place. And we don't need to always update rstats on client reply because auth MDS will have the latest rstat (either itself or synced from non-auth MDS via scatter/gather process).

So we only need 2 changes: 1) CEPH_STAT_RSTAT flag, 2) forward the request. What do you think?

Reply:
By looking through the code again, we need 3 changes: 1) CEPH_STAT_RSTAT flag, 2) forward the request, 3) always update rstat on client if reply is from auth MDS. What do you think?